src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -118,6 +118,56 @@ def _list_app(app_types, resource_group_name=None):
     return result
 
 
+def get_auth_settings(resource_group_name, name, slot=None):
+    return _generic_site_operation(resource_group_name, name, 'get_auth_settings', slot)
+
+
+def update_auth_settings(resource_group_name, name, enabled=None, action=None, aad_client_id=None,
+                         token_store=None, runtime_version=None, token_refresh_extension_hours=None,
+                         allowed_external_redirect_urls=None, aad_client_secret=None, aad_allowed_token_audiences=None,
+                         aad_token_issuer_url=None, facebook_app_id=None, facebook_app_secret=None,
+                         facebook_oauth_scopes=None, twitter_consumer_key=None, twitter_consumer_secret=None,
+                         google_client_id=None, google_client_secret=None, google_oauth_scopes=None,
+                         microsoft_account_client_id=None, microsoft_account_client_secret=None,
+                         microsoft_account_oauth_scopes=None, slot=None):
+    auth_settings = get_auth_settings(resource_group_name, name, slot)
+
+    if action == 'AllowAnonymous':
+        auth_settings.unauthenticated_client_action = UnauthenticatedClientAction.allow_anonymous
+    elif action:
+        auth_settings.unauthenticated_client_action = UnauthenticatedClientAction.redirect_to_login_page
+        auth_settings.default_provider = AUTH_TYPES[action]
+
+    auth_attr_map = {
+        'enabled': enabled == 'true',
+        'token_store_enabled': token_store == 'true',
+        'runtime_version': runtime_version,
+        'allowed_external_redirect_urls': allowed_external_redirect_urls,
+        'token_refresh_extension_hours': token_refresh_extension_hours,
+        'client_id': aad_client_id,
+        'client_secret': aad_client_secret,
+        'allowed_audiences': aad_allowed_token_audiences,
+        'issuer': aad_token_issuer_url,
+        'facebook_app_id': facebook_app_id,
+        'facebook_app_secret': facebook_app_secret,
+        'facebook_oauth_scopes': facebook_oauth_scopes,
+        'twitter_consumer_key': twitter_consumer_key,
+        'twitter_consumer_secret': twitter_consumer_secret,
+        'google_client_id': google_client_id,
+        'google_client_secret': google_client_secret,
+        'google_oauth_scopes': google_oauth_scopes,
+        'microsoft_account_client_id': microsoft_account_client_id,
+        'microsoft_account_client_secret': microsoft_account_client_secret,
+        'microsoft_account_oauth_scopes': microsoft_account_oauth_scopes
+    }
+    for attr, arg in auth_attr_map.items():
+        if arg:
```

Comment:
This might not work all the time because a falsey value will never be set, say when you disable the auth. Suggest cross check how `update_site_configs` accomplishes the similar goal